runtime/compiler/codegen/J9CodeGenerator.cpp

```diff
@@ -3137,21 +3137,53 @@ static void processAOTGuardSites(TR::CodeGenerator *cg, uint32_t inlinedCallSize
       }
    }
 
+static void addInlinedSiteRelocation(TR::CodeGenerator *cg,
+                                     TR_ExternalRelocationTargetKind reloType,
+                                     uint8_t *reloLocation,
+                                     int32_t inlinedSiteIndex,
+                                     TR::SymbolReference *callSymref,
+                                     TR_OpaqueClassBlock *receiver,
+                                     uint8_t *destinationAddress)
+   {
+   TR_ASSERT_FATAL(reloType != TR_NoRelocation, "TR_NoRelocation specified as reloType for inlinedSiteIndex=%d, reloLocation=%p, callSymref=%p, receiver=%p",
+                   inlinedSiteIndex, reloLocation, callSymref, receiver);
+
+   TR_RelocationRecordInformation *info = (TR_RelocationRecordInformation *)cg->comp()->region().allocate(sizeof(TR_RelocationRecordInformation));
```

Comment:
better not have virtual methods (i'm sure it doesn't, but...)

Reply:
you mean `allocate()`? It doesn't, but if it did, would it be a problem? All other parts of the code would also be using the same `allocate()` right?

Reply:
i meant `TR_RelocationRecordInformation`. Wouldn't it be better to initialize this structure with a placement new, which would also have the benefit of initializing any vft needed? I'm not that concerned about it, but it's not a completely outrageous evolution someone could dream up :) .